rust-runtime/aws-smithy-http-server/src/routing/mod.rs

```diff
@@ -35,11 +38,12 @@ pub use self::{into_make_service::IntoMakeService, route::Route};
 #[derive(Debug)]
 pub struct Router<B = Body> {
     routes: Vec<Route<B>>,
+    request_specs: Vec<RequestSpec>,
```

Comment:
I'd prefer it if `routes` became a vector of pairs `(RequestSpec, Route<B>)`. And the looping code gets simplified, no `.get()`s and matches.

Reply:
Yeah, it's a better approach, you are right.